pkg/tcpip/network/fragmentation/reassembler.go

```diff
@@ -78,9 +80,10 @@ func (r *reassembler) updateHoles(first, last uint16, more bool) bool {
 	return used
 }
 
-func (r *reassembler) process(first, last uint16, more bool, vv buffer.VectorisedView) (buffer.VectorisedView, bool, int, error) {
+func (r *reassembler) process(first, last uint16, more bool, vv buffer.VectorisedView, stack *stack.Route, netHeader buffer.View) (buffer.VectorisedView, bool, int, error) {
 	r.mu.Lock()
 	defer r.mu.Unlock()
+	timer := time.NewTimer(DefaultReassembleTimeout)
```

Comment:
This will create a new timer everytime process is called. That is probably not what you want. Ideally we create the timer when the reassembler is created?

Reply:
We have removed the start of timer from reassembler function.
Now we start the timer at the beginning of the fragmentation process.